convert2rhel/unit_tests/pkghandler_test.py

```diff
@@ -241,14 +249,68 @@ def __call__(self, patterns=None):
             pkg_obj.name = "installed_pkg"
             return [pkg_obj]
 
-    @unit_tests.mock(yum.rpmsack.RPMDBPackageSack, "returnPackages",
-                     ReturnPackagesMocked())
-    def test_get_installed_pkg_objects(self):
+    class QueryMocked(unit_tests.MockFunction):
+        def __init__(self):
+            self.filter_called = 0
+            self.installed_called = 0
+            self.stop_iteration = False
+            self.pkg_obj = None
+
+        def __call__(self, *args):
+            self._setup_pkg()
+            return self
+
+        def __iter__(self):
+            return self
```

Comment:
@SpyTec, do you remember based on what did you create this `QueryMocked` object? dnf uses this library https://github.com/rpm-software-management/hawkey/blob/master/src/query.c which is installed on the CentOS system as a compiled shared object.
Is the __iter__ even necessary?

Reply:
I know `__iter__` was there due to the test failing, since you need both `__iter__` and `__next__` for an iterator. I'm not sure if that is still the case but I would presume it does. 

Since this is in a test file I don't see a reason not to add a pylint exception though